test/unit/matrix/test_tile.cpp

```diff
@@ -341,3 +339,300 @@ TYPED_TEST(TileTest, CreateBuffer) {
+
+  ASSERT_TRUE(tile_f.is_ready());
+  EXPECT_FALSE(next_tile_f.is_ready());
+  // check tile pointer and unlock next_tile.
+  // next_tile_f should be ready.
+  checkFullTile(tile_ptr, tile_f.get(), size);
+
+  ASSERT_TRUE(next_tile_f.is_ready());
+  checkFullTile(tile_ptr, next_tile_f.get(), size);
+}
+
+template <class T, Device D>
+void testSubtileDisjoint(std::string name, TileElementSize size, SizeType ld,
+                         const std::vector<SubTileSpec>& specs, std::size_t last_dep) {
+  SCOPED_TRACE(name);
+  if (specs.size() > 0) {
+    ASSERT_LT(last_dep, specs.size());
+  }
+
+  auto tmp = createTileAndPtrChecker<T, D>(size, ld);
+  auto tile = std::move(std::get<0>(tmp));
+  auto tile_ptr = std::move(std::get<1>(tmp));
+
+  hpx::lcos::local::promise<Tile<T, D>> tile_p;
+  hpx::future<Tile<T, D>> tile_f;
+  hpx::future<Tile<T, D>> next_tile_f;
+  std::tie(tile_p, tile_f, next_tile_f) = createTileChain<T, D>();
+  ASSERT_TRUE(tile_f.valid() && !tile_f.is_ready());
+  ASSERT_TRUE(next_tile_f.valid() && !next_tile_f.is_ready());
+
+  // create subtiles
+  auto subtiles = splitTileDisjoint(tile_f, specs);
+  ASSERT_TRUE(tile_f.valid());
+  ASSERT_FALSE(tile_f.is_ready());
+  ASSERT_EQ(specs.size(), subtiles.size());
+
+  checkValidNonReady(subtiles);
+  ASSERT_FALSE(tile_f.is_ready());
+  ASSERT_FALSE(next_tile_f.is_ready());
+
+  // Make subtiles ready and check them
+  tile_p.set_value(std::move(tile));
+
+  if (subtiles.size() > 0) {
+    checkReadyAndDependencyChain(tile_ptr, subtiles, specs, last_dep, tile_f);
+  }
+  ASSERT_TRUE(tile_f.is_ready());
+  EXPECT_FALSE(next_tile_f.is_ready());
+  // check tile pointer and unlock next_tile.
+  // next_tile_f should be ready.
+  checkFullTile(tile_ptr, tile_f.get(), size);
+
+  ASSERT_TRUE(next_tile_f.is_ready());
+  checkFullTile(tile_ptr, next_tile_f.get(), size);
+}
+
+TYPED_TEST(TileTest, Subtile) {
+  using Type = TypeParam;
+
+  testSubtile<Type, Device::CPU>("Test 1", {5, 7}, 8, {{3, 4}, {2, 3}});
+  testSubtile<Type, Device::CPU>("Test 2", {5, 7}, 8, {{4, 6}, {1, 1}});
+  testSubtile<Type, Device::CPU>("Test 3", {5, 7}, 8, {{0, 0}, {5, 7}});
+
+  testSubtileDisjoint<Type, Device::CPU>("Test Vector Empty", {5, 7}, 8, {}, 0);
+  testSubtileDisjoint<Type, Device::CPU>("Test Vector 1", {5, 7}, 8, {{{3, 4}, {2, 3}}}, 0);
+  testSubtileDisjoint<Type, Device::CPU>("Test Vector 2", {5, 7}, 8,
+                                         {{{4, 3}, {0, 0}}, {{4, 6}, {1, 1}}}, 1);
+  testSubtileDisjoint<Type, Device::CPU>("Test Vector 3", {5, 7}, 8,
+                                         {{{5, 7}, {0, 0}},
+                                          {{1, 2}, {2, 2}},
+                                          {{3, 0}, {2, 7}},
+                                          {{0, 0}, {1, 4}},
+                                          {{0, 4}, {3, 3}}},
+                                         2);
+  testSubtileDisjoint<Type, Device::CPU>("Test Vector 3", {5, 7}, 8,
```

Comment:
minor: I think it's just a message for reference in case of errors, right?
```suggestion
  testSubtileDisjoint<Type, Device::CPU>("Test Vector 4", {5, 7}, 8,
```